### PR TITLE
57 61 63 64 score events and bot not found

### LINKED ISSUE #57: No error when the bot executable is not found
When the runner tries to run the sample java bot, there is no visible error when it fails to find the actual executable jar file. This makes it really hard to realize the actual issue. 
For example, with the change in version number, the java jar file produced by the build is `java-sample-bot-1.1-SNAPSHOT-jar-with-dependencies.jar` instead of the `java-sample-bot-1.0-SNAPSHOT-jar-with-dependencies.jar`. The bot.json file still pointed at the old version number.
When I ran the `run.bat` command there was no indication from the runner that it was failing to locate the defined jar file. Instead it just complains that no command file is present. I would have expected a big error message saying the jar file is missing or at least that the java command it must be running internally failed.

### LINKED ISSUE #61: Add roundIncomeEnergy for textMap.txt
https://forum.entelect.co.za/t/missing-data-in-textmap-txt/390

### LINKED ISSUE #63: Better unit testing for scoring events
Some scoring events do not match up with the rules. These results needs to be investigated

- Damage points is given on all missiles that hit a building for example if a building has hp of 5 and 2 missiles hit it at the same time the player gets 10 points but only one missile hits the building, the other continues and will possible get the player 5 more points.
- Damage points is given to a player for missiles hitting his own building, these missiles don’t do any damage to the building though

### LINKED ISSUE #64: Add buildingType to Building Data in TextMap.txt
Add the building type to the building data in the TextMap.txt.

### PR DESCRIPTION
**Pull request by Blaarkies**
Fix #57 Bot not found exception
Fix #61 roundIncomeEnergy for textMap.txt
Fix #63 score events not working
Fix #64 buildingType to Building Data in TextMap.txt

### CODE REVIEW

File: game-engine/core/src/main/java/za/co/entelect/challenge/core/engine/TowerDefenseRoundProcessor.java [hunk 1]
```diff
@@ -103,24 +103,26 @@ private static boolean positionMatch(Cell a, Cell b) {
...
+                                                    && !b.isPlayers(missile.getPlayerType())
+                                                    && b.getHealth() > 0)
+                                            .forEach(b -> {
+                                                TowerDefensePlayer missileOwner = null;
+                                                try {
+                                                    missileOwner = towerDefenseGameMap.getPlayer(missile.getPlayerType());
+                                                } catch (Exception e) {
+                                                    e.printStackTrace();
+                                                }
+                                                b.damageSelf(missile, missileOwner);
```
Comment (Blaarkies) [inline game-engine/core/src/main/java/za/co/entelect/challenge/core/engine/TowerDefenseRoundProcessor.java]: && _!b.isPlayers(missile.getPlayerType())_
building is not owned by the missile owner

_&& b.getHealth() > 0_
only check alive buildings

File: game-engine/domain/src/main/java/za/co/entelect/challenge/entities/Building.java [hunk 2]
```diff
@@ -76,19 +76,16 @@ public Building getInvertedXInstance() {
...
 
-            if (health <= 0) {
-                health = 0;
-            }
+        missileOwner.addScore(damageTaken * destroyMultiplier);
 
-            incomingMissile.setSpeed(0);
+        if (damageTaken > 0) {
+            m.setSpeed(0);
         }
```
Comment (Blaarkies) [inline game-engine/domain/src/main/java/za/co/entelect/challenge/entities/Building.java]: Similar logic used in determining how much damage player health takes when homebase is hit

File: game-engine/domain/src/main/java/za/co/entelect/challenge/entities/TowerDefenseGameMap.java [hunk 3]
```diff
@@ -118,14 +120,29 @@ public void removeMissile(Missile missile) {
...
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            TowerDefensePlayer missileOwner = null;
+            try {
+                missileOwner = getPlayer(p.playerType);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
```
Comment (DaanKeun) [inline game-engine/domain/src/main/java/za/co/entelect/challenge/entities/TowerDefenseGameMap.java]: Not sure if this is a bad suggestion, but can we not merge these two try catch statements. Seems pretty redundant, but then again it probably was done.

I can see that you might want missleOwner to still initialize regardless of whether opponent gets initialized, but looking at the rest of the method, if opponent is broken, missleOwner is pointless.

Comment (Blaarkies) [reply game-engine/domain/src/main/java/za/co/entelect/challenge/entities/TowerDefenseGameMap.java]: Yes that is correct. My plan is to get very descriptive exception messages in the log files when the game-engine messes up. I was thinking with these we can have distinct message depending on what happened

Comment (DaanKeun) [reply game-engine/domain/src/main/java/za/co/entelect/challenge/entities/TowerDefenseGameMap.java]: Makes sense 👍

**Pull request merged**